realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1749,8 +1749,12 @@ public Date maximumDate(String fieldName) {
      */
     public long count() {
         realm.checkIfValid();
-
-        return this.query.count();
+        // The fastest way of doing `count()` is going through `TableQuery.count()`. Unfortunately
+        // doing this does not correctly apply all side effects of queries (like subscriptions). Also
```

Comment:
> ... does not correctly apply all side effects of queries (like subscriptions)

the new implementation doesn't create a subscription either?

Reply:
You are right, but if we decided to change that in the future going directly to `TableQuery` would not work correctly.